### PR TITLE
storage promises managing nfs mounts should now correctly mount after editing fstab entries

### PR DESCRIPTION
**Pull request by olehermanse**
Intention of code seemed to be to track number of changes to fstab, and then
in the end, perform mounting if there were more than 1 change.

### CODE REVIEW

**Comment by olehermanse** [comment]
@cf-bottom jenkins

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/5279/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/5279/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-5279/

**Comment by nickanderson** [comment]
Looks like https://tracker.mender.io/browse/CFE-1864

File: cf-agent/nfs.c [hunk 1]
```diff
@@ -496,25 +495,27 @@ int VerifyInFstab(EvalContext *ctx, char *name, const Attributes *a, const Promi
 
     Log(LOG_LEVEL_VERBOSE, "Verifying '%s' in '%s'", mountpt, VFSTAB[VSYSTEMHARDCLASS]);
 
+    int changes = 0;
```
Comment (vpodzime) [inline cf-agent/nfs.c:498]: I'd prefer moving this near the top of the function and then doing `return changes` in all `return` places.

File: cf-agent/nfs.c [hunk 2]
```diff
@@ -534,7 +535,7 @@ int VerifyNotInFstab(EvalContext *ctx, char *name, const Attributes *a, const Pr
 
     host = a->mount.mount_server;
     mountpt = name;
-
+    int changes = 0;
```
Comment (vpodzime) [inline cf-agent/nfs.c:538]: Same here.

File: cf-agent/nfs.c (hunk 1 above)
Comment (olehermanse) [reply cf-agent/nfs.c:498]: Thanks for feedback, I will keep it in mind for the next time, but don't think it's big enough to do another amend + force push + re-review.

File: cf-agent/nfs.c (hunk 2 above)
Comment (olehermanse) [reply cf-agent/nfs.c:538]: Thanks for feedback, I will keep it in mind for the next time, but don't think it's big enough to do another amend + force push + re-review.

**Pull request merged**